src/infrastructure/startup/Server.ts

```diff
@@ -42,36 +46,79 @@ export default class Server implements ISetup {
 
     this.loadVersions(app);
 
+    // TODO: Make this redirect configurable
     app.get("/", (_: Request, res: Response) => {
       res.redirect("/api/v1");
     });
 
-    app.listen(API_PORT, () => {
-      Logger.info(`Example app listening at http://localhost:${API_PORT}`);
+    app.listen(config.api.port, () => {
+      Logger.info(
+        `Example app listening at http://localhost:${config.api.port}`
+      );
     });
   }
 
   private loadVersions(app: Application) {
     const directory = path.join(__dirname, "../../controllers");
 
+    const versions: string[] = [];
     glob.sync(`${directory}/v*`).forEach((directoryPath: string) => {
       if (!fs.lstatSync(path.resolve(directoryPath)).isDirectory()) return;
       const directories = directoryPath.split("/");
       const version = directories[directories.length - 1];
+      versions.push(version);
       const router = Router();
       app.use(`/api/${version}`, router);
 
-      this.mountControllers(directoryPath, router);
+      // TODO: add server info
+      const documentation = new OpenApiBuilder();
+      documentation.addTitle(config.api.name);
+      documentation.addVersion(version);
+      documentation.addContact({
+        name: config.maintainer.name,
+        email: config.maintainer.email,
+        url: config.maintainer.url,
+      });
+
+      documentation.addServer({
+        url: `/api/${version}`,
+        description: 'Current environment'
+      });
```

Comment:
Separate the concern of setting up the server and generating the api documentation.